### PR TITLE
fix(types): account for Queue.close(timeout)

### PR DESCRIPTION
**Pull request by ethanresnick**
These types support all four call signatures:

```
close(): Promise<void>
close(timeout: number): Promise<void>;

close(cb: () => void): void;
close(timeout: number, cb: () => void): void;
```

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -41,8 +41,9 @@ declare class BeeQueue {
   checkHealth(): Promise<BeeQueue.HealthCheckResult>;
   checkHealth(cb: (counts: BeeQueue.HealthCheckResult) => void): void;
 
-  close(): Promise<void>;
   close(cb: () => void): void;
+  close(timeout?: number): Promise<void>;
+  close(timeout: number, cb: () => void): void;
```
Comment (skeggse) [inline index.d.ts]: The `timeout` is technically optional here, so you can pass something like `close(timeoutOrNull, callback)`. I dunno if this works in typescript, though:
```suggestion
  close(timeout?: number, cb: () => void): void;
```

Comment (ethanresnick) [reply index.d.ts]: TS won't allow an optional parameter (with the `?`) before a required one, but I changed this to:

```ts
close(timeout: number | undefined | null, cb: () => void): void;
```

Comment (skeggse) [reply index.d.ts]: sgtm

**Comment by skeggse** [comment]
Released in `v1.2.3`.

**Pull request merged**